### DISCUSSION TITLE
Minio iOS SDK

### DISCUSSION THREAD

**Discussion by multinerd** [minio/minio#14449] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
With AWS S3 deprecation of path styles and the lack of a ios client for minio, i can no longer upload/download files as the awssdk resolves to `https://bucket.minio.domain.tld`

**Describe the solution you'd like**
an ios sdk would be amazing.

**Describe alternatives you've considered**
I've considered setting up dns records for each bucket, but i would have to maintain that.
Currently, I am using a (much older) modified version of aws sdk that forces path style but this should only be a workaround.

**Additional context**

**Comment by harshavardhana** [comment]
@multinerd we would love to write one but unfortunately, we do not have the bandwidth or the time for the time being. 

Closing this since this may not happen in near future.

**Comment by multinerd** [comment, marked as answer]
That's unfortunate.

My workaround (if anyone else needs it)

https://github.com/aws-amplify/aws-sdk-ios/blob/main/AWSCore/Utility/AWSCategory.m#L454

```objc
- (BOOL)aws_isVirtualHostedStyleCompliant {
    return NO;
}
```

**Reply by yuxiaoleigithub** [reply]
After using your method, the “uploadFile” method can be uploaded, but the “uploadUsingMultiPart“ method will still report an AccessDenied error. How can I solve this, please?

**Reply by yuxiaoleigithub** [reply]
@multinerd   Hello, I really need your help, thank you

**Reply by yuxiaoleigithub** [reply]
My email address is”yuxiaolei542@126.com“, looking forward to your reply. thank you @multinerd
